Client/Frontend/Browser/LocalRequestHelper.swift

```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import WebKit
+
+class LocalRequestHelper: BrowserHelper {
+    func scriptMessageHandlerName() -> String? {
+        return "localRequestHelper"
+    }
+
+    func userContentController(userContentController: WKUserContentController, didReceiveScriptMessage message: WKScriptMessage) {
+        guard message.frameInfo.request.URL?.isLocal ?? false else { return }
+
+        let params = message.body as! [String: String]
+
+        if params["type"] == "load",
+           let url = NSURL(string: params["url"]!) {
```

Comment:
I think this should be an `if let` .. we cannot trust data coming in via JS. Better to ignore bad URLs than to crash on them.